Sources/Purchasing/Configuration.swift

```diff
@@ -145,8 +153,21 @@ import Foundation
          * - Warning: This assumes your IAP implementation uses StoreKit 1.
          * Observer mode is not compatible with StoreKit 2.
          */
+        @available(*, deprecated, message: "Use with(purchasesAreCompletedBy:) instead.")
         @objc public func with(observerMode: Bool) -> Configuration.Builder {
-            self.observerMode = observerMode
+            self.purchasesAreCompletedBy = observerMode ? .myApp : .revenueCat
+            return self
+        }
+        /**
+         * Set `purchasesAreCompletedBy`.
+         * - Parameter purchasesAreCompletedBy: Set this to `.myApp` if you have your own IAP implementation and
+         * want to use only RevenueCat's backend. Default is `.revenueCat`.
+         *
+         * - Warning: This assumes your IAP implementation uses StoreKit 1.
+         * `.myApp` is not compatible with StoreKit 2.
```

Comment:
??

Reply:
Why not? Your app would always be able to finish, right? Wouldn't it be the other way around, if there are compatibility issues at all?

Reply:
I reused the documentation from `observerMode:` - is it incorrect or outdated? It was added in September 2023.

Reply:
Yeah, that'll be outdated, we managed to make it work in v5. 

Which... maybe we should just keep this new thing in v5, and have this PR point to the 5.0.0-dev branch? @fire-at-will @MarkVillacampa thoughts?